### PR TITLE
ES Modules

### PR DESCRIPTION
**Pull request by amanmahajan7**
//  TODO

### CODE REVIEW

File: packages/react-data-grid/index.js [hunk 1]
```diff
@@ -1 +1,3 @@
-module.exports = require('./src');
\ No newline at end of file
+import Grid from './src';
+export * from './src';
+export default Grid;
```
Comment (amanmahajan7) [inline packages/react-data-grid/index.js]: We need to fix publishing before we publish the next version
https://github.com/adazzle/react-data-grid/blob/next/ci/publish/replacePackageEntry.js#L7

File: packages/react-data-grid-examples/src/__tests__/GridRunner.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
-const React        = require('react');
-const ReactDOM = require('react-dom');
-const TestUtils    = require('react-dom/test-utils');
-const { Component: ExampleGrid } = require('../scripts/example14-all-features-immutable');
+import React from 'react';
+import ReactDOM from 'react-dom';
+import TestUtils from 'react-dom/test-utils';
 import { mount } from 'enzyme';
 
+const { Component: ExampleGrid } = require('../scripts/example14-all-features-immutable');
```
Comment (MayhemYDG) [inline packages/react-data-grid-examples/src/__tests__/GridRunner.js]: You've missed this one.

File: packages/react-data-grid/index.js [hunk 3]
```diff
@@ -1 +1,3 @@
-module.exports = require('./src');
\ No newline at end of file
+import Grid from './src';
```
Comment (MayhemYDG) [inline packages/react-data-grid/index.js]: You can simplify
```js
export { default } from './src';
export * from './src';
```

File: packages/react-data-grid/src/ColumnMetrics.js [hunk 4]
```diff
@@ -137,12 +136,12 @@ function compareEachColumn(prevColumns, nextColumns, isSameColumn) {
...
+export function sameColumns(prevColumns, nextColumns, isSameColumn) {
   if (areColumnsImmutable(prevColumns, nextColumns)) {
     return prevColumns === nextColumns;
   }
 
   return compareEachColumn(prevColumns, nextColumns, isSameColumn);
 }
 
-module.exports = { recalculate, resizeColumn, sameColumn, sameColumns };
+export { sameColumn };
```
Comment (MayhemYDG) [inline packages/react-data-grid/src/ColumnMetrics.js]: This is just a re-export, is it necessary to have it here?

File: packages/react-data-grid/src/index.js [hunk 5]
```diff
@@ -1,17 +1,20 @@
 import Grid from './ReactDataGrid';
-import RowComparer from 'common/utils/RowComparer';
-import Cell from './Cell';
-import Row from './Row';
-import EmptyChildRow from './EmptyChildRow';
+export default Grid;
```
Comment (MayhemYDG) [inline packages/react-data-grid/src/index.js]:
```js
export { default } from './ReactDataGrid';
```

File: packages/react-data-grid/src/ColumnMetrics.js (hunk 4 above)
Comment (amanmahajan7) [reply packages/react-data-grid/src/ColumnMetrics.js]: I need to revisit the entire `ColumnMetrics.js` logic and we can refine it then. Currently we are using `ColumnMetrics.sameColumns` in a few places

**Pull request merged**